exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2155 @@
+		return nil, err
+	}
+	timer := time.NewTimer(ok.WebsocketResponseMaxLimit)
+	wsResponse := make(chan *wsIncomingData)
+	ok.WsResponseMultiplexer.Register <- &wsRequestInfo{
+		ID:   randomID,
+		Chan: wsResponse,
+	}
+	defer func() { ok.WsResponseMultiplexer.Unregister <- randomID }()
+	for {
+		select {
+		case data := <-wsResponse:
+			if data.Operation == okxOpBatchAmendOrders && data.ID == input.ID {
+				if data.Code == "0" || data.Code == "2" {
+					resp, err := data.copyToPlaceOrderResponse()
+					if err != nil {
+						return nil, err
+					}
+					return resp.Data, nil
+				}
+				if len(data.Data) == 0 {
+					return nil, fmt.Errorf("error code:%s message: %v", data.Code, ErrorCodes[data.Code])
+				}
+				var resp WsOrderActionResponse
+				dataByte, err := json.Marshal(data)
+				if err != nil {
+					return nil, err
+				}
+				err = json.Unmarshal(dataByte, &resp)
+				if err != nil {
+					return nil, err
+				}
+				errs := common.Errors{}
+				for x := range resp.Data {
+					if resp.Data[x].SCode != "0" {
+						errs = append(errs, fmt.Errorf("error code:%s message: %v", resp.Data[x].SCode, resp.Data[x].SMessage))
+					}
+				}
+				return nil, errs
+			}
+			continue
+		case <-timer.C:
+			timer.Stop()
+			return nil, fmt.Errorf("%s websocket connection: timeout waiting for response with an operation: %v",
+				ok.Name,
+				input.Operation)
+		}
+	}
+}
+
+// Run this functions distributes websocket request responses to
+func (m *wsRequestDataChannelsMultiplexer) Run() {
+	ticker := time.NewTicker(time.Second)
+	for {
+		select {
+		case <-ticker.C:
+			for x, myChan := range m.WsResponseChannelsMap {
+				if myChan == nil {
+					delete(m.WsResponseChannelsMap, x)
+				}
+			}
+		case id := <-m.Unregister:
+			delete(m.WsResponseChannelsMap, id)
+		case reg := <-m.Register:
+			m.WsResponseChannelsMap[reg.ID] = reg
+		case msg := <-m.Message:
+			if msg.ID != "" && m.WsResponseChannelsMap[msg.ID] != nil {
+				m.WsResponseChannelsMap[msg.ID].Chan <- msg
+				continue
+			}
+			for _, myChan := range m.WsResponseChannelsMap {
+				if msg.Event == "error" || myChan.Event == "login" &&
+					(msg.Code == "60009" || msg.Code == "60022" || msg.Code == "60018") &&
+					strings.Contains(msg.Msg, myChan.Channel) {
+					myChan.Chan <- msg
+					continue
+				} else if msg.Event != myChan.Event ||
+					msg.Argument.Channel != myChan.Channel ||
+					msg.Argument.InstrumentType != myChan.InstrumentType ||
+					msg.Argument.InstrumentID != myChan.InstrumentID {
+					continue
+				}
+				myChan.Chan <- msg
+				break
+			}
+		}
+	}
+}
+
+// WsChannelSubscription send a subscription or unsubscription request for different channels through the websocket stream.
+func (ok *Okx) WsChannelSubscription(operation, channel string, assetType asset.Item, pair currency.Pair, tooglers ...bool) (*SubscriptionOperationResponse, error) {
+	if !(strings.EqualFold(operation, "subscribe") || strings.EqualFold(operation, "unsubscribe")) {
```

Comment:
```suggestion
	if !strings.EqualFold(operation, "subscribe") && !strings.EqualFold(operation, "unsubscribe") {
```
Does this need to be equalfold?